includes/class-wcpdf-main.php

```diff
@@ -1582,29 +1582,31 @@ public function wc_webhook_trigger( $document, $order ) {
 	 * @return void
 	 */
 	public function display_due_date( string $document_type, $order ): void {
-		if ( 'invoice' !== $document_type || empty( $order ) ) {
+		if ( empty( $order ) ) {
 			return;
 		}
 
-		$invoice = wcpdf_get_invoice( $order );
+		$document = wcpdf_get_document( $document_type, $order );
 
-		if ( ! $invoice ) {
+		if ( ! $document ) {
 			return;
 		}
 
-		$due_date_timestamp = $invoice->get_due_date();
+		$due_date_timestamp = is_callable( array( $document, 'get_due_date' ) ) ? $document->get_due_date() : 0;
 
 		if ( 0 >= $due_date_timestamp ) {
 			return;
 		}
 
-		$due_date = apply_filters( 'wpo_wcpdf_invoice_due_date_display', date( wcpdf_date_format( $this, 'due_date' ), $due_date_timestamp ), $due_date_timestamp, $order );
+		$due_date          = apply_filters( "wpo_wcpdf_{$document->slug}_due_date_display", date( wcpdf_date_format( $this, 'due_date' ), $due_date_timestamp ), $due_date_timestamp, $order, $document_type );
+		$document_settings = $document->get_settings();
+		$due_date_label    = ! empty( $document_settings['due_date_label'] ) ? $document_settings['due_date_label'] : __( 'Due Date:', 'woocommerce-pdf-invoices-packing-slips' );
```

Comment:
I think that the `$document_settings['due_date_label']` must be able to be translated

Reply:
I'm not sure I understood. Do you mean the 'Due Date:' string, which is already translatable?

Reply:
No, I mean the one that you fill in the field.

Reply:
@alexmigf If I understood the implementation correctly, the other part of the translation should be in the pro extension, in the `filter_document_titles`(). Am I right?

Reply:
I think you just need to wrap the saved value under `__()`? But I haven't tested.

Reply:
Do you mean I should have to wrap `$document_settings['due_date_label']` under `__()`?

By the way, I added the below code in the `filter_document_titles()` method in the pro extension and tested it with Polylang. It was working correctly:
```php
add_filter( "wpo_wcpdf_{$_document->slug}_due_date_title", function ( $title, $document = null ) use ( $_document ) {
	return $this->get_custom_title_for_filter( 'due_date_title', $title, $document, $_document );
}, 10, 2 );
```

Reply:
Hum, maybe the label field should be added only in Pro like the others, and in the free we use the filter. Now that you mentioned the `filter_document_titles()` I saw that we have all title fields in Pro only. Sorry for the trouble :(